collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -162,10 +162,12 @@ public void moveForward(HashMap<FormIndex, IAnswerData> answers, Boolean evaluat
         isLoading.setValue(true);
 
         scheduler.immediate((Supplier<Boolean>) () -> {
-            return updateAnswersForScreen(answers, evaluateConstraints);
+            return saveScreenAnswersToFormController(answers, evaluateConstraints);
         }, updateSuccess -> {
             isLoading.setValue(false);
 
+            formController.getAuditEventLogger().flush();
```

Comment:
Move these `flush` calls to the `foreground` block so they run on the UI thread (preventing concurrent `logEvent`/`flush` calls).

Reply:
How did that happen? Were you able to reproduce? I;m thinking whether we should improve AuditEventLogger to protect it from cases like that instead.

Reply:
Wasn't able to reproduce, but I could see that we'd moved a `flush` call onto the background in v2022.3.0. I do agree we should make it "thread safe" eventually (I comment about that [here](https://github.com/getodk/collect/pull/5259#discussion_r952459314)), but I think this is the smaller change we can make to prevent the crash.

Reply:
For a hotfix ok but please file a separate issue because it's not a solution I love.